lib/chewy/strategy/delayed_sidekiq.rb

```diff
@@ -0,0 +1,185 @@
+# frozen_string_literal: true
+
+require_relative '../index'
+
+# patch Chewy::Index with a delayed sidekiq options method
+# example usage:
+#   class UsersIndex < Chewy::Index
+#     delayed_sidekiq_options latency: 10, margin: 2, reindex_wrapper: ->(&reindex) {
+#       ActiveRecord::Base.connected_to(role: :reading) do
+#         reindex.call
+#       end
+#     }
+#     ...
+#   end
+Chewy::Index.define_singleton_method(:delayed_sidekiq_options) do |opts = {}|
+  @delayed_sidekiq_options ||= Struct.new(:latency, :margin, :reindex_wrapper, keyword_init: true).new(
+    latency: opts[:latency] || Chewy::Config.instance.configuration.dig(:delayed_sidekiq, :latency),
+    margin: opts[:margin] || Chewy::Config.instance.configuration.dig(:delayed_sidekiq, :margin),
+    reindex_wrapper: ->(&reindex) { reindex.call }
+  )
+end
+
+# patch Chewy::Index import method with :strategy option
+# example usage:
+#   UsersIndex.import([user1.id], update_fields: [:email], strategy: :delayed_sidekiq)
+Chewy::Index.define_singleton_method(:import_routine) do |*args|
+  *ids, options = args
+  if options.is_a?(Hash) && options.delete(:strategy) == :delayed_sidekiq
+    return if ids.empty?
+
+    return 'delayed_sidekiq supports ids only!' unless ids.all? do |id|
+      id.respond_to?(:to_i)
+    end
+
+    begin
+      Chewy::Strategy::DelayedSidekiq::Scheduler.new(self, ids, options).postpone
+    rescue StandardError => e
+      e.message
+    else
+      return # to match super behaviour - return nothing (means no errors)
+    end
+  else
+    super(*args)
+  end
+end
```

Comment:
Why do you prefer to monkeypath when you can edit the files directly?

Reply:
No problem. I can easily add these changes to Chewy::Index if you are OK with that.
I copied that stuff from the `patch file` I have in the project so that is why everything exists in the single file.